docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,382 @@
 
-let fetchHtmlAsync url =
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
+    }
+
+[<EntryPoint>]
+let main argv =
+    countFileBytes "path-to-file.txt"
+    |> Async.RunSynchronously
 
-        // Execution of fetchHtmlAsync won't continue until the result
-        // of AsyncDownloadString is bound.
-        let! html = webClient.AsyncDownloadString(uri)
-        return html
+    Console.Read() |> ignore
+    0 // return an integer exit code
+```
+
+In the example, the `countFileBytes` function is of type `string -> Async<unit>`. Calling the function does not actually execute the asynchronous computation. Instead, it returns an `Async<unit>` that acts as a *specification* of the work that is to execute asynchronously. It calls `Async.AwaitTask` in its body, which will convert the result of <xref:System.IO.File.WriteAllBytesAsync%2A> to an appropriate type when it is called.
+
+Another important line is the call to `Async.RunSynchronously`. This is one of the async starting functions that you'll need to call if you want to actually execute an F# asynchronous computation.
+
+This is a key difference with the C#/VB style of `async` programming. In F#, asynchronous computations can be thought of as **Cold tasks**. They must be explicitly started to actually execute. This has some advantages, as it allows you to combine and sequence asynchronous work much more easily than in C#/VB.
+
+## Combining asynchronous computations
+
+Here is an example that builds upon the previous one by combining computations:
+
+```fsharp
+open System
+open System.IO
+
+let countFileBytes path =
+    async {
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
     }
 
-let html = "https://dotnetfoundation.org" |> fetchHtmlAsync |> Async.RunSynchronously
-printfn "%s" html
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Parallel
+    |> Async.Ignore
+    |> Async.Start
+
+    Console.Read() |> ignore
+    0 // return an integer exit code
 ```
 
-And that’s it! Aside from the use of `async`, `let!`, and `return`, this is just normal F# code.
+As you can see, the `main` function has quite a few more calls made. Conceptually, it does the following:
 
-There are a few syntactical constructs which are worth noting:
+1. Transform the command-line arguments into `Async<unit>` computations with `Array.map`
+2. Create an `Async<'T[]>` that schedules and runs the `countFileBytes` computations in parallel when it runs.
+3. Create an `Async<unit>` that will run the parallel computation and ignore its result.
+4. Explicitly run the last computation with `Async.Start`.
```

Comment:
worth mentioning that it kicks them off but does not await and that maybe what will take place if you press a key. (Or consider using StartAsChild for more easy to talk about semantics?)